src/platform/Zephyr/WiFiManager.cpp

```diff
@@ -53,11 +75,38 @@ CHIP_ERROR WiFiManager::Init()
         ChipLogError(DeviceLayer, "wpa_supplicant is not initialized!");
         return CHIP_ERROR_INTERNAL;
     }
-    else
-    {
-        ChipLogDetail(DeviceLayer, "wpa_supplicant has been initialized");
+
+    // TODO: consider moving these to ConnectivityManagerImpl to be prepared for handling multiple interfaces on a single device.
+    Inet::UDPEndPointImplSockets::SetJoinMulticastGroupHandler([](Inet::InterfaceId interfaceId, const Inet::IPAddress & address) {
+        const in6_addr addr = ToZephyrAddr(address);
+        net_if * iface      = GetInterface(interfaceId);
+        VerifyOrReturnError(iface != nullptr, INET_ERROR_UNKNOWN_INTERFACE);
+
+        net_if_mcast_addr * maddr = net_if_ipv6_maddr_add(GetInterface(interfaceId), &addr);
```

Comment:
We can use  `iface` which was already checked against the `nullptr` instead of calling `GetInterface(interfaceId)` again, right?

Reply:
Right, thanks!

Reply:
Done